deps/generate.jl

```diff
@@ -71,87 +72,105 @@ function generate_jls(
 end
 
 """
-Parse a `sourcefile` (.agf) into a native dictionary, where each `kvp = (glassname, glassinfo)` is a glass.
+Parse a `sourcefile` (.agf) into a native Dict, `catalogdict`, where each `kvp = (glassname, glassinfo)` is a glass.
+
+| 1  | 2        | 3         | 4     | 5     | 6                     | 7             | 8         | 9         | 10    | 11    |
+|:---|:---------|:----------|:------|:------|:----------------------|:--------------|:----------|:----------|:------|:------|
+| NM | raw name | dispform  | ?     |  Nd   | Vd                    | [exclude_sub  | status    | meltfreq] |
+| ED | TCE      | ?         | p     |  ΔPgF | [ignore_thermal_exp]  |
+| CD | C1       | C2        | C3    |  C4   | C5                    | C6            | C7        | C8        | C9    | C10   |
+| TD | D₀       | D₁        | D₂    |  E₀   | E₁                    | λₜₖ           | temp      |
+| OD | relcost  | CR        | FR    |  SR   | AR                    | PR
+| LD | λmin     | λmax      |
+| IT | T1       | T2        | T3    |
 """
-function sourcefile_to_catalog(source_file::AbstractString)
-    catalog_dict = Dict{String,Dict{String}}()
+function sourcefile_to_catalog(sourcefile::AbstractString)
+    catalogdict = Dict{String,Dict{String}}()
 
-    # check whether the file is UTF8 or UTF16 encoded
-    is_utf8 = isvalid(readuntil(source_file, " "))
-    fo = is_utf8 ? open(source_file, "r") : open(source_file, enc"UTF-16LE", "r")
```

Comment:
@friggog, I'm using DelimitedFiles.readdlm to parse in the .AGF file. Not sure if it'll run into problems with UTF16 encoding - is there a file I can test this on?

Reply:
Test fail answered this for me - OHARA.agf :(

Working on a fix